code/CloudFlare_Purge.php

```diff
@@ -0,0 +1,531 @@
+
+    /**
+     * @var
+     */
+    protected $response;
+
+    /**
+     * @var array
+     */
+    protected $fileTypes = array(
+        'image' => array(
+            "bmp" ,"gif" ,"jpg" ,"jpeg" ,"pcx" ,"tif" ,"png" ,"alpha","als" ,"cel" ,"icon" ,"ico" ,"ps", "svg"
+        ),
+        'javascript' => array(
+            "js"
+        ),
+        'css' => array(
+            'css', 'cssmap'
+        )
+    );
+
+    /**
+     * @var string
+     */
+    protected static $endpoint = "https://api.cloudflare.com/client/v4/zones/:identifier/purge_cache";
+
+    /**
+     * @param array $files
+     * @return $this
+     */
+    public function setFiles(array $files)
+    {
+        $this->clearFiles();
+        $this->pushFile($files);
+
+        return $this;
+    }
+
+    /**
+     * @param string $file
+     * @return $this
+     */
+    public function pushFile($file)
+    {
+        if (!is_array($this->files)) {
+            $this->files = array();
+        }
+
+        if (is_array($file)) {
+            foreach ($file as $pointer) {
+                $this->pushFile($pointer);
+            }
+
+            return $this;
+        }
+
+        array_push($this->files, $this->convertToAbsolute($file));
+
+        return $this;
+    }
+
+    /**
+     * Recursively find files with a specific extension(s) starting at the document root
+     *
+     * @param string|array $extensions
+     *
+     * @param null|string $dir A directory relevant to the project root, if null the entire project root will be search
+     * @return $this
+     */
+    public function findFilesWithExts($extensions, $dir = null)
+    {
+        $files = array();
+        $rootDir = str_replace('//', '/', $_SERVER['DOCUMENT_ROOT'] . Director::baseURL() . "/" . $dir);
+
+        if (is_array($extensions)) {
+            foreach ($extensions as $ext) {
+                $files = array_merge($this->rglob(rtrim($rootDir, "/") . "/*.{$ext}"), $files);
+            }
+        }
+
+        if (is_string($extensions)) {
+            $files = $this->rglob(rtrim($rootDir, "/") . "/*{$extensions}");
+        }
+
+        $this->pushFile($files);
+
+        return $this;
+    }
+
+    /**
+     * Recursive Glob Function
+     *
+     * @param     $pattern
+     * @param int $flags
+     *
+     * @return array
+     */
+    private function rglob($pattern, $flags = 0)
+    {
+        $files = glob($pattern, $flags);
+        foreach (glob(dirname($pattern) . '/*', GLOB_ONLYDIR | GLOB_NOSORT) as $dir) {
+            if (basename($dir) == 'framework') {
+                continue;
+            }
+            // if we implode the array we should be able to search for all extensions at once like below
+            // http://stackoverflow.com/a/23969253/2266583 w/ GLOB_BRACE
+            // however it would always return no files for me, so i'll just leave it here for now
+            // $files = array_merge($this->rglob(rtrim($rootDir, "/") . '/*.{'.$extensions.'}'), $files);
```

Comment:
This is definitely a priority, the more extensions to search for the longer the request takes.

Windows 10 WAMP PHP 5.5 - Unable to get any results from the GLOB_BRACE approach